app/cmd/doctor.go

```diff
@@ -0,0 +1,189 @@
+		jenkinsServer := jenkinsServers[i]
+		jclient.URL = jenkinsServer.URL
+		jclient.UserName = jenkinsServer.UserName
+		jclient.Token = jenkinsServer.Token
+		jclient.Proxy = jenkinsServer.Proxy
+		jclient.ProxyAuth = jenkinsServer.ProxyAuth
+		outString += "  checking the No." + strconv.Itoa(i) + " - " + jenkinsServer.Name + " status: "
+		if _, err := jclient.GetPlugins(); err == nil {
+			outString += "***available***\n"
+		} else {
+			outString += "***unavailable*** " + err.Error() + "\n"
+		}
+	}
+	err = errors.New(outString)
+	return
+}
+
+func checkCurrentPlugins(jclient *client.PluginManager) (err error){
+	outString := "Begining to checking the current JenkinsServer's plugins status: \n"
+	getCurrentJenkinsAndClient(&jclient.JenkinsCore)
+	if plugins, err := jclient.GetPlugins(2); err == nil {
+		if err = cyclePlugins(plugins); err != nil {
+			outString += err.Error()
+		}
+	} else {
+		outString += "  No plugins have lost dependencies...\n"
+	}
+	err = errors.New(outString)
+	return
+}
+
+// cyclePlugins is check all installed plugins
+func cyclePlugins(plugins *client.InstalledPluginList) (err error){
+	outString := ""
+	for _, plugin := range plugins.Plugins {
+		outString += "  Checking the plugin " + plugin.ShortName + ": \n"
+		dependencies := plugin.Dependencies
+		if len(dependencies) != 0 {
+			if err = cycleDependencies(dependencies, plugins); err != nil {
+				outString += err.Error()
+			}
+		} else {
+			outString += "    The Plugin no dependencies\n"
+		}
+	}
+	err = errors.New(outString)
+	return
+}
+
+func cycleDependencies(dependencies []client.Dependence, plugins *client.InstalledPluginList) (err error){
+	outString := ""
+	for _, dependence := range dependencies {
+		outString += "    Checking the dependence plugin " + dependence.ShortName + ": "
+		hasInstalled := false
+		needUpdate := false
+		if err = cycleMatchPlugins(plugins, dependence, hasInstalled, needUpdate); err != nil {
+			outString += err.Error()
+		}
+	}
+	err = errors.New(outString)
+	return
+}
+
+func cycleMatchPlugins(plugins *client.InstalledPluginList, dependence client.Dependence, hasInstalled bool, needUpdate bool) (err error){
+	outString := ""
+	for _, checkPlugin := range plugins.Plugins {
+		checkPluginVersion := strings.Split(checkPlugin.Version, ".")
+		dependenceVersion := strings.Split(dependence.Version, ".")
+		if checkPlugin.ShortName == dependence.ShortName {
+			hasInstalled = true
+			if _, err = matchPlugin(dependenceVersion, checkPluginVersion, needUpdate, dependence);err != nil {
+				outString += err.Error()
+			}
+
+		}
+		if needUpdate {
+			break
+		}
+	}
+	if !hasInstalled {
+		outString += "\n    The dependence " + dependence.ShortName + " no install, please install it the version " + dependence.Version + " at least\n"
+	}
+	err = errors.New(outString)
+	return
+}
+
+func matchPlugin(dependenceVersion []string, checkPluginVersion []string, needUpdate bool, dependence client.Dependence) (isPass bool, err error) {
```

Comment:
Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed.
See also https://sonarcloud.io/project/issues?branch=pr-169&id=jenkins-zh_jenkins-cli&open=AW411QeU_XpztwYnd3Zt&resolved=false&types=CODE_SMELL